src/main/java/yevhen/bookstore/dto/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,24 @@
+package yevhen.bookstore.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import yevhen.bookstore.validation.FieldMatch;
+
+@FieldMatch
+public record UserRegistrationRequestDto(
+        @NotBlank
+        @Email
```

Comment:
It's better to check which regexp is used in @Email validation. Maybe it's better to create custom one with validator